src/main/java/org/allenai/wikitables/WikiTablesExample.java

```diff
@@ -4,41 +4,85 @@
 import com.jayantkrish.jklol.nlpannotation.AnnotatedSentence;
+import edu.stanford.nlp.sempre.ContextValue;
+import edu.stanford.nlp.sempre.Formula;
+import edu.stanford.nlp.sempre.Value;
 
 /**
- * Similar to CcgExample, except that this supports multiple logcal forms per example.
- * This correctly does not hold a syntactic parse or dependencies. But we may need them in the future.
+ * Similar to CcgExample, except that this supports multiple logical forms per example, and also stores the
+ * context and target values from Sempre to execute and evaluate formulas.
+ * This currently does not hold a syntactic parse or dependencies. But we may need them in the future.
  */
 
-public class WikiTablesExample extends CcgExample {
+public class WikiTablesExample {
 
-  public final Set<Expression2> logicalForms;
-  public WikiTablesExample(AnnotatedSentence sentence, Set<Expression2> logicalForms) {
-    // We are making the logicalForm in CcgExample null, and defining the set logicalForms in this class.
-    super(sentence, null, null, null);
+  private final AnnotatedSentence sentence;
+  private final Set<Expression2> logicalForms;
+  private final ContextValue context;
+  private final Value targetValue;
+
+  public WikiTablesExample(AnnotatedSentence sentence, Set<Expression2> logicalForms, ContextValue context,
+                           Value targetValue) {
+    this.sentence = sentence;
     this.logicalForms = logicalForms;
+    this.context = context;
+    this.targetValue = targetValue;
   }
 
   public Set<Expression2> getLogicalForms() {
     return logicalForms;
   }
 
-  @Override
-  public boolean hasLogicalForm() {
-    return logicalForms != null;
+  public AnnotatedSentence getSentence() {
+    return sentence;
+  }
+
+  public boolean isFormulaCorrect(Expression2 pnpFormula) {
+    // Sempre represents lambda expressions differently. We changed them when reading the examples. Changing
+    // them back for execution.
+    System.out.println("Before conversion: " + pnpFormula.toString());
+    String expressionString = toSempreLambdaForm(pnpFormula.toString());
+    System.out.println("After conversion: " + expressionString);
+    try {
+      Formula sempreFormula = Formula.fromString(expressionString);
+      return WikiTablesDataProcessor.isFormulaCorrect(sempreFormula, context, targetValue, null);
+    } catch (Exception ex) {
+      System.err.println("Bad formula: " + expressionString);
+      return false;
+    }
   }
 
-  @Override
   public String toString() {
-    String superString = super.toString();
     List<String> logicalFormStrings = new ArrayList<>();
     for (Expression2 lf : logicalForms) {
         logicalFormStrings.add(lf.toString());
     }
-    return superString + " [\n" + String.join(" ", logicalFormStrings) + "\n]";
+    return sentence + " [\n" + String.join(" ", logicalFormStrings) + "\n]";
+  }
+
+  public static String toPnpLambdaForm(String expression) {
```

Comment:
Let's extract the logical form conversion methods to separate class containing only static utility functions, e.g., WikiTablesUtil.

Also, please don't do these conversions using regexes -- both of these methods are clearly broken if the variables have different names.